core/domain/prod_validation_jobs_one_off_test.py

```diff
@@ -13062,6 +13062,68 @@ def test_missing_edited_exploration_model_failure(self):
         run_job_and_check_output(self, expected_output, sort=True)
 
 
+class UserAuthModelValidatorTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(UserAuthModelValidatorTests, self).setUp()
+
+        self.signup(USER_EMAIL, USER_NAME)
+        self.user_id = self.get_user_id_from_email(USER_EMAIL)
+        self.gae_id = self.get_gae_id_from_email(USER_EMAIL)
+        user_models.UserAuthModel(id=self.user_id, gae_id=[self.gae_id]).put()
+        self.model_instance = user_models.UserAuthModel.get_by_id(
+            self.user_id)
+        self.job_class = (
+            prod_validation_jobs_one_off
+            .UserAuthModelAuditOneOffJob)
+
+    def test_standard_operation(self):
+        expected_output = [
+            u'[u\'fully-validated UserAuthModel\', 1]']
+        run_job_and_check_output(self, expected_output)
+
+    def test_model_with_created_on_greater_than_last_updated(self):
+        self.model_instance.created_on = (
+            self.model_instance.last_updated + datetime.timedelta(days=1))
+        self.model_instance.put()
+        expected_output = [(
+            u'[u\'failed validation check for time field relation check '
+            'of UserAuthModel\', '
+            '[u\'Entity id %s: The created_on field has a value '
+            '%s which is greater than the value '
+            '%s of last_updated field\']]') % (
+                self.user_id, self.model_instance.created_on,
+                self.model_instance.last_updated
+            )]
+        run_job_and_check_output(self, expected_output)
+
+    def test_model_with_last_updated_greater_than_current_time(self):
+        expected_output = [(
+            u'[u\'failed validation check for current time check of '
+            'UserAuthModel\', '
+            '[u\'Entity id %s: The last_updated field has a '
+            'value %s which is greater than the time when the job was run\']]'
+        ) % (self.user_id, self.model_instance.last_updated)]
+
+        with self.swap(datetime, 'datetime', MockDatetime13Hours), self.swap(
+            db.DateTimeProperty, 'data_type', MockDatetime13Hours):
+            update_datastore_types_for_mock_datetime()
+            run_job_and_check_output(self, expected_output)
+
+    def test_missing_user_settings_model_failure(self):
```

Comment:
Per https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests point 2, all test names should include the conditions being tested & the expected outcomes. Please update the new test names in this test suite to match.

Reply:
I knew about this, but actually I was following the naming convention as rest of the test classes did. The 4 tests that I wrote in addition to setup are written in every other class with this name only. Do you still want me to update their names (If we would be updating names for other tests too in future) ? 
@ankita240796 your thoughts? 
I am asking this because validation tests don't have a defined action - validation itself in the action, right? Condition is also not defined since we testing that particular model only. Outcome is the thing which is only defined in all the test names.

Reply:
@sajalasati let us update the test names as suggested by @BenHenning - it is better if we do not introduce more error cases. We are also planning a refactor of prod jobs, so the existing ones will be fixed then.

Reply:
@ankita240796 Thanks for commenting on that. 
Can you provide an example how the renaming would look like? 
Say on the test function - `test_model_with_last_updated_greater_than_current_time`. As I mentioned before I'm having a hard time understanding here the 2 things - action and withCondition. The outcome part is clear which is failure.

Reply:
Maybe something along the line of: test_audit_with_last_updated_greater_than_current_time_fails

Reply:
Done now.